apps/ejabberd/src/mod_mam_riak_timed_arch_yz.erl

```diff
@@ -174,8 +174,16 @@ lookup_messages_muc(Result, Host,
                     IsSimple).
 
 
-archive_size(Size, _Host, _ArchiveID, _ArchiveJID) ->
-    Size.
+archive_size(_Size, _Host, _ArchiveID, ArchiveJID) ->
+    OwnerJID = ArchiveJID,
+    RemoteJID = undefined,
+    {MsgIdStartNoRSM, MsgIdEndNoRSM} =
+        calculate_msg_id_borders(undefined, undefined, undefined, undefined),
+    F = fun get_msg_id_key/3,
+    {TotalCount, _} = read_archive(OwnerJID, RemoteJID,
```

Comment:
I just implemented it using the same function, that is used to count messages in offset and index functions.
I guess we should edit them if we want to optimize something.
Also, I don't care about performance here: it's used from tests only.